src/DependencyInjection/SonataUserExtension.php

```diff
@@ -66,15 +67,7 @@ public function load(array $configs, ContainerBuilder $container): void
 
         $loader->load('form.xml');
 
-        if (class_exists('Google\Authenticator\GoogleAuthenticator')) {
-            @trigger_error(
-                'The \'Google\Authenticator\' namespace is deprecated in sonata-project/GoogleAuthenticator since version 2.1 and will be removed in 3.0.',
-                E_USER_DEPRECATED
-            );
-        }
-
-        if (class_exists('Google\Authenticator\GoogleAuthenticator') ||
-            class_exists('Sonata\GoogleAuthenticator\GoogleAuthenticator')) {
+        if (class_exists(GoogleAuthenticator::class)) {
             $loader->load('google_authenticator.xml');
```

Comment:
couldn't we do the method call `$this->configureGoogleAuthenticator($config, $container);` here instead of line 105 and remove the second check for the class inside the method itself?

Reply:
not sure it is the good way as it is a require-dev dependency then i think first part of `configureGoogleAuthenticator` method has to be ran even if dependency is not installed on no dev context. WDYT?